mta/fixtures/application_inventory.py

```diff
@@ -3,50 +3,36 @@
 import pytest
 
 from swagger_client.models.api_application import ApiApplication
-from swagger_client.models.api_task import ApiTask
 
 
-@pytest.fixture(scope="session")
 def json_application():
     with open("mta/data/application.json", "r") as file:
-        yield json.load(file)
+        return json.load(file)
 
 
 @pytest.fixture(scope="session")
-def analyses_data():
+def applications_data():
+    return json_application()
+
+
+def json_analysis():
     with open("mta/data/analysis.json", "r") as file:
         return json.load(file)
 
 
+@pytest.fixture(params=json_analysis(), ids=[item.get("source") for item in json_analysis()])
+def analysis_item(request):
+    # Fixture that returns each analysis_item from the list of analysis.json
+    return request.param
+
+
 @pytest.fixture(scope="session")
-def applications(source_username_credentials, json_application, create_api, delete_api):
-    apps = []
-    for app in json_application:
-        application_data = {}
-        for key, value in app.items():
-            application_data[key] = value
-        if "identities" in application_data:
-            application_data["identities"] = [{"id": source_username_credentials.id}]
-        api_app = ApiApplication(**application_data)
-        new_app = create_api.applications_post(api_app.to_dict())
-        apps.append(new_app)
-    yield apps
-    for app in apps:
-        delete_api.applications_id_delete(app.id)
-
-
-@pytest.fixture()
-def tasks(applications, analyses_data, create_api, get_api, update_api):
-    tasks_from_db = []
-    for app in applications:
-        app_id = app.id
-        app_name = app.name
-        if app_name in analyses_data:
-            analysis_info = analyses_data[app_name]
-            task_data = {"targets": analysis_info["targets"], "output": "/windup/report"}
-            application = {"id": app_id, "name": app_name}
-            task = ApiTask(addon="windup", application=application, data=task_data)
-            new_task = create_api.tasks_post(task=task.to_dict())
-            update_api.tasks_id_submit_put(id=new_task.id, task=new_task)
-            tasks_from_db.append(get_api.tasks_id_get(new_task.id))
-    return tasks_from_db
+def application(source_username_credentials, applications_data, create_api, delete_api, request):
+    app_name = request.param
+    app_data = applications_data[app_name]
+    if "identities" in app_data:
```

Comment:
maybe i missed it, but i do not see any app with identities
that means this is block is not tested.
maybe add it later, when you add and app with identities?

Reply:
@amastbau please see application.json file - in case the app dictionary includes `"identities": []` - I add the creds when creating the app